anon_enum.md

```diff
@@ -27,6 +27,31 @@ match int_from_file() {
 }
 ```
 
+It is also possible to merge multiple types that implements the same trait into a new one.
+```rust
+let some_data = vec![
+    vec![1, 2, 3],
+    vec![4, 5, 6],
+];
+let new_data: Vec<usize> = some_data
+    .iter()
+    .map(|line| -> enum impl Iterator {
+        // This will create an iterator from the union of two different itertor types
+        if some_condition {
+            line.iter() as enum
+        } else {
+            line.iter().rev as enum
+        }
+    })
+    .flat_map()
```

Comment:
You probably mean to replace the `map` with `flat_map` above, and to replace `rev` with `rev()`.

Reply:
Yes, thanks.